app/modules/intelligence/agents/custom_agents/custom_agents_service.py

```
@@ -776,8 +826,8 @@ async def fetch_available_tools(self, user_id: str) -> List[str]:
         try:
             tools = ToolService(self.db, user_id).list_tools()
             return [tool.id for tool in tools]
-        except Exception as e:
-            logger.error(f"Error fetching available tools: {str(e)}")
+        except Exception:
+            logger.exception("Error fetching available tools", user_id=user_id)
             raise HTTPException(
                 status_code=500, detail="Failed to fetch available tools"
             )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)